### PR TITLE
Add LazyBlock javadoc regarding lazy block loader

### PR DESCRIPTION
**Pull request by sopel39**

### CODE REVIEW

File: presto-spi/src/main/java/io/prestosql/spi/block/LazyBlock.java [hunk 1]
```diff
@@ -30,6 +30,10 @@
 
     private Block block;
 
+    /**
+     * Creates a {@link LazyBlock} instance. Lazy block loader
+     * should load block partially if possible.
+     */
```
Comment (dain) [inline presto-spi/src/main/java/io/prestosql/spi/block/LazyBlock.java]: I'd put the documentation about the loader on the `LazyBlockLoader` interface

Comment (sopel39) [reply presto-spi/src/main/java/io/prestosql/spi/block/LazyBlock.java]: done. Does `LazyBlockLoader` still need to accept `T`?

Comment (dain) [reply presto-spi/src/main/java/io/prestosql/spi/block/LazyBlock.java]: I can't remember why I designed the API this way, and I find it really confusing.  I feel like we should be able to replace `LazyBlockLoader` with `Suppler<Block>`

File: presto-spi/src/main/java/io/prestosql/spi/block/LazyBlockLoader.java [hunk 2]
```diff
@@ -15,5 +15,8 @@
 
 public interface LazyBlockLoader<T extends Block>
 {
+    /**
+     * Loads a lazy block. Lazy block loader should load block partially if possible.
```
Comment (findepi) [inline presto-spi/src/main/java/io/prestosql/spi/block/LazyBlockLoader.java]: Can you elaborate what "partially" means here?

**Pull request merged**